### PR TITLE
feature(redis) sentinel support

### LINKED ISSUE #160: Do code change to make sure harbor operator support reading sentinel mode configurations from spec

### PR DESCRIPTION
**Pull request by bitsf**
close #160 
feature(redis) sentinel support

### CODE REVIEW

**Comment by bitsf** [comment]
## current core
_REDIS_URL_REG=redis://:diwhfnwr@harborcluster-sample-redis/2

_REDIS_URL=harborcluster-sample-redis:6379,100,diwhfnwr,0,30

## target core
_REDIS_URL_REG=redis+sentinel://:diwhfnwr@rfs-harborcluster-sample-redis/mymaster/1

_REDIS_URL_CORE=redis+sentinel://:diwhfnwr@rfs-harborcluster-sample-redis/mymaster/0

## current jobservice
config.yml
```
worker_pool:
  backend: "redis"

  redis_pool:
    redis_url: redis://:edjihfbi@harborcluster-sample-redis/1
    namespace: jobservice::cluster-sample-ns::harborcluster-sample-harbor-harbor
    idle_timeout_second: 30
```

## target jobservice
config.yml
```
worker_pool:
  backend: "redis"

  redis_pool:
    redis_url: redis+sentinel://:edjihfbi@rfs-harborcluster-sample-redis/mymaster/2z?idle_timeout_seconds=30
    namespace: jobservice::cluster-sample-ns::harborcluster-sample-harbor-harbor
    idle_timeout_second: 30
```

## current registry
/etc/registry/config.yml
```
redis:
  addr: "harborcluster-sample-redis:6379"
  # password: setted through environment variable
  db: 1
  pool:
    maxidle: 3
    maxactive: 5
    idletimeout: 30s
```

## target registry
```
redis:
  # sentinel hosts with comma
  addr: rfs-harborcluster-sample-redis:26379
  sentinelMasterSet: mymaster
  readtimeout: 10s
  writetimeout: 10s
  dialtimeout: 10s
  password: 123
  db: 1
```

File: config/config/assets/registry-config.yaml.tmpl [hunk 1]
```diff
@@ -206,6 +206,9 @@ middleware:
 {{- end }}
 {{- if .Spec.Redis }}
 redis:
+  {{- if .Spec.Redis.SentinelMasterSet }}
+  sentinelMasterSet: mymaster
```
Comment (steven-zou) [inline config/config/assets/registry-config.yaml.tmpl:210]: Here should be a variable, correct?

**Review by steven-zou — CHANGES_REQUESTED** [review]
do changes?

**Review by chlins — APPROVED** [review]
LGTM

**Review by steven-zou — APPROVED** [review]
Approved

**Pull request merged**